packages/dev/serializers/src/glTF/2.0/glTFMaterialExporter.ts

```diff
@@ -84,240 +64,89 @@ function getFileExtensionFromMimeType(mimeType: ImageMimeType): string {
+        return pixels;
+    } else {
+        throw new Error("Unsupported pixel format!");
+    }
+}
 
-        const diffuse = babylonStandardMaterial.diffuseColor.toLinearSpace(babylonStandardMaterial.getScene().getEngine().useExactSrgbConversions).scale(0.5);
-        const opacity = babylonStandardMaterial.alpha;
-        const specularPower = Scalar.Clamp(babylonStandardMaterial.specularPower, 0, _GLTFMaterialExporter._MaxSpecularPower);
-
-        const roughness = solveForRoughness(specularPower);
-
-        const glTFPbrMetallicRoughness: IMaterialPbrMetallicRoughness = {
-            baseColorFactor: [diffuse.r, diffuse.g, diffuse.b, opacity],
-            metallicFactor: 0,
-            roughnessFactor: roughness,
-        };
+/**
+ * Utility methods for working with glTF material conversion properties.
+ * @internal
+ */
+export class GLTFMaterialExporter {
+    private _exporter: GLTFExporter;
 
-        return glTFPbrMetallicRoughness;
-    }
+    // Mapping to store textures
+    private _textureMap = new Map<BaseTexture, ITextureInfo>();
 
-    /**
-     * Computes the metallic factor
-     * @param diffuse diffused value
-     * @param specular specular value
-     * @param oneMinusSpecularStrength one minus the specular strength
-     * @returns metallic value
-     */
-    public static _SolveMetallic(diffuse: number, specular: number, oneMinusSpecularStrength: number): number {
-        if (specular < this._DielectricSpecular.r) {
-            this._DielectricSpecular;
-            return 0;
-        }
+    // Mapping of internal textures to images to avoid exporting duplicate images
+    private _internalTextureToImage: { [uniqueId: number]: { [mimeType: string]: Promise<number> } } = {};
 
-        const a = this._DielectricSpecular.r;
-        const b = (diffuse * oneMinusSpecularStrength) / (1.0 - this._DielectricSpecular.r) + specular - 2.0 * this._DielectricSpecular.r;
-        const c = this._DielectricSpecular.r - specular;
-        const D = b * b - 4.0 * a * c;
-        return Scalar.Clamp((-b + Math.sqrt(D)) / (2.0 * a), 0, 1);
+    constructor(exporter: GLTFExporter) {
+        this._exporter = exporter;
     }
 
-    /**
-     * Sets the glTF alpha mode to a glTF material from the Babylon Material
-     * @param glTFMaterial glTF material
-     * @param babylonMaterial Babylon material
-     */
-    private static _SetAlphaMode(glTFMaterial: IMaterial, babylonMaterial: Material & { alphaCutOff: number }): void {
-        if (babylonMaterial.needAlphaBlending()) {
-            glTFMaterial.alphaMode = MaterialAlphaMode.BLEND;
-        } else if (babylonMaterial.needAlphaTesting()) {
-            glTFMaterial.alphaMode = MaterialAlphaMode.MASK;
-            glTFMaterial.alphaCutoff = babylonMaterial.alphaCutOff;
-        }
+    public getTextureInfo(babylonTexture: Nullable<BaseTexture>): Nullable<ITextureInfo> {
+        return babylonTexture ? this._textureMap.get(babylonTexture) || null : null;

```

Comment:
When trying to check for null/undefined, it's more robust to use `??`
```suggestion
        return babylonTexture ? this._textureMap.get(babylonTexture) ?? null : null;
```